nautobot/extras/filters.py

```diff
@@ -429,6 +431,24 @@ def search(self, queryset, name, value):
         return queryset.filter(Q(name__icontains=value) | Q(user__username__icontains=value))
 
 
+class ScheduledJobFilterSet(BaseFilterSet):
+    q = django_filters.CharFilter(
+        method="search",
+        label="Search",
+    )
+    first_run = django_filters.DateTimeFilter()
+    last_run = django_filters.DateTimeFilter()
+
+    class Meta:
+        model = ScheduledJob
+        fields = ["id", "first_run", "last_run", "total_run_count"]
+
+    def search(self, queryset, name, value):
+        if not value.strip():
+            return queryset
+        return queryset.filter(Q(name__icontains=value) | Q(job_class__icontains=value) | Q(task__icontains=value))
```

Comment:
Should we also search in the `description`, or is that overkill?

Is the `task` user-facing? If not, maybe exclude it from the filter?